staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/status.go

```diff
@@ -35,14 +35,19 @@ func ErrorToAPIStatus(err error) *metav1.Status {
 	switch t := err.(type) {
 	case statusError:
 		status := t.Status()
-		if len(status.Status) == 0 {
-			status.Status = metav1.StatusFailure
-		}
-		if status.Code == 0 {
-			switch status.Status {
-			case metav1.StatusSuccess:
+		switch status.Status {
+		case metav1.StatusSuccess:
+			if status.Code == 0 {
```

Comment:
why was this made conditional? this was always set to 200 before

Reply:
@liggitt in the earlier incarnation, `status.Code = http.StatusOK` was called only if `status.Code == 0` and `status.Status` is set to `metav1.StatusSuccess` ... no?

Reply:
oh, right...